### PR TITLE
Don't write IP/MAC to device running newer driver version 4.1 or newer

### PR DESCRIPTION
**Pull request by longlimsft**
Since RDMA driver 4.1 we are moving to a different framework to probe ND device. The device for exposing MAC/IP is removed from kernel and no longer needed.

### CODE REVIEW

File: azurelinuxagent/common/rdma.py [hunk 1]
```diff
@@ -199,10 +199,24 @@ def start(self):
...
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False;
+            retcode,out = shellutil.run_get_output("modinfo hv_network_direct")
+            if retcode == 0 :
+                version = re.search("version:\s+(\d+)\.(\d+)\.(\d+)\D", out, re.IGNORECASE)
+                if version :
+                    v1 = version.groups(0)[0]
+                    v2 = version.groups(0)[1]
+                    if int(v1+v2)>40 :
```
Comment (ahmetalpbalkan) [inline azurelinuxagent/common/rdma.py]: what's this 40 @longlimsft ?

Comment (ahmetalpbalkan) [inline azurelinuxagent/common/rdma.py]: if you're trying to check v>4.1 this is not a good way to do it. v0.101 will pass this check too

File: azurelinuxagent/common/rdma.py [hunk 2]
```diff
@@ -199,10 +199,24 @@ def start(self):
...
         try:
             RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
-            RDMADeviceHandler.wait_rdma_device(
-                self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
-            RDMADeviceHandler.write_rdma_config_to_device(
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False;
+            retcode,out = shellutil.run_get_output("modinfo hv_network_direct")
+            if retcode == 0 :
```
Comment (ahmetalpbalkan) [inline azurelinuxagent/common/rdma.py]: extra spaces before `:`s in your patch

File: azurelinuxagent/common/rdma.py [hunk 3]
```diff
@@ -199,10 +199,24 @@ def start(self):
     def process(self):
         try:
             RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
-            RDMADeviceHandler.wait_rdma_device(
-                self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
-            RDMADeviceHandler.write_rdma_config_to_device(
-                self.rdma_dev, self.ipv4_addr, self.mac_addr)
+
+            skip_rdma_device = False;
```
Comment (ahmetalpbalkan) [inline azurelinuxagent/common/rdma.py]: some `;`s in your patch, not needed

File: azurelinuxagent/common/rdma.py (hunk 1 above)
Comment (longlimsft) [inline azurelinuxagent/common/rdma.py]: I think this should be fine, we check against 4.0. The version numbers can only go up in future.

Comment (longlimsft) [inline azurelinuxagent/common/rdma.py]: I think this should be fine, we check against 4.0. The version numbers can only go up in future.

Comment (longlimsft) [inline azurelinuxagent/common/rdma.py]: I think this should be fine, we check against 4.0. The version numbers can only go up in future.

File: azurelinuxagent/common/rdma.py (hunk 2 above)
Comment (longlimsft) [inline azurelinuxagent/common/rdma.py]: Will fix this

File: azurelinuxagent/common/rdma.py (hunk 3 above)
Comment (longlimsft) [inline azurelinuxagent/common/rdma.py]: Will fix this.

File: azurelinuxagent/common/rdma.py (hunk 1 above)
Comment (ahmetalpbalkan) [inline azurelinuxagent/common/rdma.py]: @longlimsft it's very sloppy. is it harder to write a longer if statement?

**Comment by hglkrijger** [comment]
@ahmetalpbalkan, @longlimsft - any updates on this?

**Comment by ahmetalpbalkan** [comment]
This currently needs fixes from Long.

**Comment by longlimsft** [comment]
I will request a new pull request to address comments.

**Pull request closed without merging**